lib/appsignal/absinthe.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Appsignal.Absinthe do
+  require Appsignal.Utils
+
+  @tracer Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer, Appsignal.Tracer)
+  @span Appsignal.Utils.compile_env(:appsignal, :appsignal_span, Appsignal.Span)
+
+  @moduledoc false
+
+  require Logger
+
+  def attach do
+    handlers = %{
+      [:absinthe, :execute, :operation, :start] => &__MODULE__.absinthe_execute_operation_start/4,
+      [:absinthe, :execute, :operation, :stop] => &__MODULE__.absinthe_execute_operation_stop/4
+    }
+
+    for {event, fun} <- handlers do
+      detach = :telemetry.detach({__MODULE__, event})
+      attach = :telemetry.attach({__MODULE__, event}, event, fun, :ok)
+
+      case {detach, attach} do
+        {:ok, :ok} ->
+          _ =
+            Appsignal.IntegrationLogger.debug(
+              "Appsignal.Absinthe reattached to #{inspect(event)}"
+            )
+
+          :ok
+
+        {{:error, :not_found}, :ok} ->
+          _ =
+            Appsignal.IntegrationLogger.debug("Appsignal.Absinthe attached to #{inspect(event)}")
+
+          :ok
+
+        {_, {:error, _} = error} ->
+          Logger.warn("Appsignal.Absinthe not attached to #{inspect(event)}: #{inspect(error)}")
+
+          error
+      end
+    end
+  end
+
+  def absinthe_execute_operation_start(_event, _measurements, _metadata, _config) do
+    "http_request"
+    |> @tracer.create_span(@tracer.current_span())
+    |> @span.set_name("graphql")
```

Comment:
A static "graphql" label doesn't add much information, but we could see if we can add either "query" or "mutation", possibly with the operation name, if that's available on the metadata. Setting the query on the body would be great, but probably won't fit most of the times before we truncate it.

Reply:
Agreed! Do note that queries/mutations may not have names, though. In that case just `query` or `mutation` would be fine.

Reply:
Note that this is the event name in most situations, so the name here isn't the root span name unless this is called outside of a Phoenix application. I'm fine with changing this, but unsure what you'd like me to change it to.

As for the body, we have the full query in the parameters, and the Ecto query in a nested event, which I think should be sufficient.

Reply:
In other GraphQL instrumentations (I'm only familiar with the Node.js 2.x and 3.x ones) we use the _operation type_ (`query`, `mutation`) and the _operation name_, if present, like @tombruijn described. So, for a query like:

```graphql
query HeroNameAndFriends {
  hero {
    name
    friends {
      name
    }
  }
}
```

We'd show `query HeroNameAndFriends` as the span name. This would also make it group queries in a meaningful ways in situations where the queries were root spans.

Reply:
For me it was already of interest to see this information in the performance timeline on hover. I don't think it's too relevant to see it for the incident action name.